augur/filter.py

```diff
@@ -319,9 +319,8 @@ def filter_by_date(metadata, date_column="date", min_date=None, max_date=None):
     """
     strains = set(metadata.index.values)
 
-    # Skip this filter if no valid min/max date is given or the date column does
-    # not exist.
-    if (not min_date and not max_date) or date_column not in metadata.columns:
```

Comment:
Removing this check does slightly change the function, i.e. it will now filter out strains with invalid dates even when min/max date is not provided. Maybe also update the docstring & doctests to reflect this change?

Reply:
Thinking about it some more, I think the checks for `args.min_date`/`args.max_date` and `date_column` should be done in the same place. Ideally would be [here](https://github.com/nextstrain/augur/blob/945b49af50cc72c4343673491531ae630db40040/augur/filter.py#L609), but `metadata.columns` is not available with the current implementation of `construct_filters()`.

This change was needed in the original implementation using `args.min_date_offset`/`args.max_date_offset` and optional for the current implementation.

I'll drop fbb1303169efa5853d67133b44454f81e5454b04 for now.